clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -580,6 +579,21 @@ protected Collection<PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate>> i
     return pubSubMessages;
   }
 
+  boolean maybeUpdatePartitionToBootstrapMap(
+      PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> message,
+      PubSubTopicPartition pubSubTopicPartition,
+      long startTimestamp) {
+    ControlMessage controlMessage = (ControlMessage) message.getValue().getPayloadUnion();
+    if (controlMessage.controlMessageType == START_OF_SEGMENT.getValue()
+        && Arrays.equals(message.getKey().getKey(), KafkaKey.HEART_BEAT.getKey())) {
```

Comment:
I don't think we must restrict this to HBs only... any recent-enough message should be able to kick us into the caught up state.